### ISSUE TITLE
getsockopt: connection refused

### ISSUE THREAD

**Issue by jasonknight** [buger/gor#370]
When running the linux binary as sudo on Debian 8, All my connections are refused: dial tcp 
`127.0.0.1:8080: getsockopt: connection refused`

when running command: `./gor --input-raw :8080 --output-stdout`

I get output:

`Version: v0.15.1
`

**Comment by jasonknight** [comment]
I miss understood the working of the tool, it requires you listen IN on port communication, so you have to bind to a port that some other program is already listening on. Otherwise no joy.

**Comment by buger** [comment]
Right! I guess you expected it will work like a proxy? 

I'm evaluating this option as well, in cases when traffic interception is not possible, do you think it will be useful in your case?

Thank you!

**Comment by jasonknight** [comment]
Well my particular case was this: I needed to see what the EXACT text of the request my program was sending out. I.e. I needed to double check that net/http was in fact playing fast and loose with headers (which it does, how evil). 

I'm not sure gor really was meant to solve my case directly. I just had to point my program towards my nginx instance and point gor to port 80 and vroom, all was fine. So I got the info, and gor worked perfectly once I understood how it worked...

It could be interesting as a feature, either an option --proxy-to blah.blah.blah:4004 etc, or a PRO feature... doing both recording and proxying...

I think in the end it'd probably take away from core functionality, depends whether you have the time and motivation to add it in...
